lib/calendarium-romanum/calendar.rb

```diff
@@ -126,6 +126,12 @@ def day(*args, vespers: false)
       )
     end
 
+    def each
+      (temporale.start_date..temporale.end_date)
+        .map { |date| day(date) }
```

Comment:
Both time and memory are being wasted here:

* `map` iterates over the date range, retrieves calendar data and collects them in a new array
* `each` iterates over this array and yields it's elements

Both (retrieve calendar data, yield them) should be done in a single `each`. This way you save one iteration as well as memory necessary for the temporary array.